iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractBranch.java

```diff
@@ -196,95 +196,169 @@ public boolean isOverloaded() {
-                return checkTemporaryLimits1(limitReduction);
+                return checkTemporaryLimits1(limitReduction, type);
 
             case TWO:
-                return checkTemporaryLimits2(limitReduction);
+                return checkTemporaryLimits2(limitReduction, type);
 
             default:
                 throw new AssertionError();
         }
     }
 
+    @Override
+    public Branch.Overload checkTemporaryLimits(Side side, float limitReduction) {
+        return checkTemporaryLimits(side, limitReduction, LimitType.CURRENT);
+    }
+
+    @Override
+    public Branch.Overload checkTemporaryLimits(Side side, LimitType type) {
+        return checkTemporaryLimits(side, 1f, type);
+    }
+
     @Override
     public Branch.Overload checkTemporaryLimits(Side side) {
-        return checkTemporaryLimits(side, 1f);
+        return checkTemporaryLimits(side, LimitType.CURRENT);
+    }
+
+    @Override
+    public Branch.Overload checkTemporaryLimits1(float limitReduction, LimitType type) {
+        return LimitViolationUtils.checkTemporaryLimits(this, Side.ONE, limitReduction, getValueForLimit(getTerminal1(), type), type);
     }
 
     @Override
     public Branch.Overload checkTemporaryLimits1(float limitReduction) {
-        return LimitViolationUtils.checkTemporaryLimits(this, Side.ONE, limitReduction, getTerminal1().getI());
+        return checkTemporaryLimits1(limitReduction, LimitType.CURRENT);
+    }
+
+    @Override
+    public Branch.Overload checkTemporaryLimits1(LimitType type) {
+        return checkTemporaryLimits1(1f, type);
     }
 
     @Override
     public Branch.Overload checkTemporaryLimits1() {
-        return checkTemporaryLimits1(1f);
+        return checkTemporaryLimits1(LimitType.CURRENT);
+    }
+
+    @Override
+    public Branch.Overload checkTemporaryLimits2(float limitReduction, LimitType type) {
+        return LimitViolationUtils.checkTemporaryLimits(this, Side.TWO, limitReduction, getValueForLimit(getTerminal2(), type), type);
     }
 
     @Override
     public Branch.Overload checkTemporaryLimits2(float limitReduction) {
-        return LimitViolationUtils.checkTemporaryLimits(this, Side.TWO, limitReduction, getTerminal2().getI());
+        return checkTemporaryLimits2(limitReduction, LimitType.CURRENT);
+    }
+
+    @Override
+    public Branch.Overload checkTemporaryLimits2(LimitType type) {
+        return checkTemporaryLimits2(1f, type);
     }
 
     @Override
     public Branch.Overload checkTemporaryLimits2() {
-        return checkTemporaryLimits2(1f);
+        return checkTemporaryLimits2(LimitType.CURRENT);
+    }
+
+    public double getValueForLimit(Terminal t, LimitType type) {
+        switch (type) {
+            case ACTIVE_POWER:
+                return t.getP();
+            case APPARENT_POWER:
+                return Math.sqrt(t.getP() * t.getP() + t.getQ() * t.getQ());
+            case CURRENT:
+                return t.getI();
+            case VOLTAGE:
```

Comment:
You can try t.getBusView().getBus().getV() if the bus is not null maybe...

Reply:
As discussed voltage is not (yet?) supported as a LimitType.